format/opentimestamps/ots.go

```diff
@@ -0,0 +1,123 @@
+package opentimestamps
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+func init() {
+	interp.RegisterFormat(
+		format.OpenTimestamps,
+		&decode.Format{
+			Description:  "OpenTimestamps file",
+			Dependencies: nil,
+			DecodeFn:     decodeOTSFile,
+		})
+}
+
+func decodeVarInt(d *decode.D) uint64 {
+	var value uint64 = 0
+	var shift uint64 = 0
+
+	for {
+		b := d.U8()
+		value |= (b & 0b01111111) << shift
+		shift += 7
+		if b&0b10000000 == 0 {
+			break
+		}
+	}
+
+	return value
+}
+
+var (
+	headerMagic  = []byte{0x00, 0x4f, 0x70, 0x65, 0x6e, 0x54, 0x69, 0x6d, 0x65, 0x73, 0x74, 0x61, 0x6d, 0x70, 0x73, 0x00, 0x00, 0x50, 0x72, 0x6f, 0x6f, 0x66, 0x00, 0xbf, 0x89, 0xe2, 0xe8, 0x84, 0xe8, 0x92, 0x94}
+	pendingMagic = binary.BigEndian.Uint64([]byte{0x83, 0xdf, 0xe3, 0x0d, 0x2e, 0xf9, 0x0c, 0x8e})
+	bitcoinMagic = binary.BigEndian.Uint64([]byte{0x05, 0x88, 0x96, 0x0d, 0x73, 0xd7, 0x19, 0x01})
```

Comment:
Could be written as `bitcoinMagic uint64 = 0x05_0x88_0x96_0x0d_0x73_0xd7_0x19_0x01` i think?

Reply:
I had never seen such syntax and it doesn't seem to work on my `go`.

What I really wanted was to be able to match a sequence of 8 bytes against the 8 bytes magic sequences above, but I couldn't find a way. All I could find were the uint64 matchers. The BitBufs have confused me. What am I missing?

Reply:
Ah sorry i forgot to remove some "0x", so like this (the "_":s is just for clarity)
```go
	pendingMagic uint64 = 0x83_df_e3_0d_2e_f9_0c_8e
	bitcoinMagic uint64 = 0x05_88_96_0d_73_d7_19_01
```

Yes the BitBuf stuff should mostly be used when dealing with raw bytes or > 8 bytes things, and yeah i'm not very happy with the API for it either hmm